server/application/application.go

```diff
@@ -234,9 +236,47 @@ func (s *Server) Update(ctx context.Context, q *ApplicationUpdateRequest) (*appv
 	return s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Update(a)
 }
 
-// UpdateSpec updates an application spec
+// UpdateSpec updates an application spec and filters out any invalid parameter overrides
 func (s *Server) UpdateSpec(ctx context.Context, q *ApplicationUpdateSpecRequest) (*appv1.ApplicationSpec, error) {
 	a, err := s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Get(*q.Name, metav1.GetOptions{})
+	validAppSet := make(map[string]map[string]bool)
+	for _, p := range a.Status.Parameters {
```

Comment:
Do we want to extract this into the Util class since it's the same code on server and cli side?

Reply:
if we put this into CheckValidParam than we have to loop through it for every parameter, I put it here so we would only have to loop through it once per UpdateSpec. rather than once per parameter. 

I could make a Util that is "convertToMap" that does the loop as a separate function.